rclcpp/include/rclcpp/time_source.hpp

```diff
@@ -118,6 +124,11 @@ class TimeSource
   RCLCPP_PUBLIC
   ~TimeSource();
 
+protected:
+  // Dedicated thread for clock subscription.
+  bool use_clock_thread_;
+  std::thread clock_executor_thread_;
```

Comment:
Just curious why is this protected while other new things are private? It's not a problem, just curious if it was done intentionally for some reason.

Reply:
Nevermind, I see this is for testing.